python/src/graphql.rs

```diff
@@ -484,93 +533,62 @@ impl PyRunningRaphtoryServer {
             handler
                 .sender
                 .send(BridgeCommand::StopServer)
                 .expect("Failed when sending cancellation signal");
             Ok(())
-        })
-    }
-
-    /// Wait until server completion.
-    pub(crate) fn wait(mut slf: PyRefMut<Self>, py: Python) -> PyResult<()> {
-        let server = &mut slf.server_handler;
+        })?;
+        let server = &mut self.server_handler;
         py.allow_threads(|| wait_server(server))
     }
+}
 
-    /// Wait for the server to be online.
-    ///
-    /// Arguments:
-    ///   * `timeout_millis`: the timeout in milliseconds (default 5000).
-    fn wait_for_online(&self, timeout_millis: Option<u64>) -> PyResult<()> {
-        self.apply_if_alive(|handler| handler.client.wait_for_online(timeout_millis))
-    }
-
-    /// Make a graphQL query against the server.
-    ///
-    /// Arguments:
-    ///   * `query`: the query to make.
-    ///   * `variables`: a dict of variables present on the query and their values.
-    ///
-    /// Returns:
-    ///    The `data` field from the graphQL response.
-    fn query(
-        &self,
-        py: Python,
-        query: String,
-        variables: Option<HashMap<String, PyObject>>,
-    ) -> PyResult<HashMap<String, PyObject>> {
-        self.apply_if_alive(|handler| handler.client.query(py, query, variables))
+#[pymethods]
+impl PyRunningRaphtoryServer {
+    pub(crate) fn get_client(&self) -> PyResult<PyRaphtoryClient> {
+        self.apply_if_alive(|handler| Ok(handler.client.clone()))
     }
 
-    /// Send a graph to the server.
-    ///
-    /// Arguments:
-    ///   * `name`: the name of the graph sent.
-    ///   * `graph`: the graph to send.
-    ///
-    /// Returns:
-    ///    The `data` field from the graphQL response after executing the mutation.
-    fn send_graph(
-        &self,
-        py: Python,
-        name: String,
-        graph: MaterializedGraph,
-    ) -> PyResult<HashMap<String, PyObject>> {
-        self.apply_if_alive(|handler| handler.client.send_graph(py, name, graph))
+    /// Stop the server and wait for it to finish
+    pub(crate) fn stop(mut slf: PyRefMut<Self>, py: Python) -> PyResult<()> {
+        slf.stop_server(py)
     }
+}
 
-    /// Set the server to load all the graphs from its path `path`.
-    ///
-    /// Arguments:
-    ///   * `path`: the path to load the graphs from.
-    ///   * `overwrite`: whether or not to overwrite existing graphs (defaults to False)
-    ///
-    /// Returns:
-    ///    The `data` field from the graphQL response after executing the mutation.
-    #[pyo3(signature=(path, overwrite = false))]
-    fn load_graphs_from_path(
-        &self,
-        py: Python,
-        path: String,
-        overwrite: bool,
-    ) -> PyResult<HashMap<String, PyObject>> {
-        self.apply_if_alive(|handler| handler.client.load_graphs_from_path(py, path, overwrite))
+fn is_online(url: &String) -> PyResult<bool> {
+    match reqwest::blocking::get(url) {
```

Comment:
.filter(|r| r.status().as_u16() == 200).unwrap_or(false)

Reply:
the return type is just `bool` since you are handling any errors by returning `false`